zookeeper-server/src/main/java/org/apache/zookeeper/cli/SyncCommand.java

```diff
@@ -55,18 +61,27 @@ public CliCommand parse(String[] cmdArgs) throws CliParseException {
     @Override
     public boolean exec() throws CliException {
         String path = args[1];
+        CompletableFuture<Integer> cf = new CompletableFuture<>();
+
         try {
             zk.sync(path, new AsyncCallback.VoidCallback() {
-
                 public void processResult(int rc, String path, Object ctx) {
-                    out.println("Sync returned " + rc);
+                    cf.complete(rc);
                 }
             }, null);
+
+            try {
+                int resultCode = cf.get(CONNECTION_TIMEOUT, TimeUnit.MILLISECONDS);
+                out.println("Sync returned " + resultCode);
+            } catch (TimeoutException ex) {
+                out.println("Sync is timeout within " +  CONNECTION_TIMEOUT + " ms");
```

Comment:
Should we throw an error?
The command may not have succeeded

Reply:
@eolivelli  pardon me.I may not get your idea clearly.
Do you mean that throwing this `TimeoutException`?

Reply:
@maoling yes

Reply:
Does this code need it's own 'try' block?  Can this catch be moved to the outer 'try' statement?